### PR TITLE
ACL migration - cmdlets to migrate share level permissions from on-prem share to share on Azure

### PR DESCRIPTION
**Pull request by chdevala**

### CODE REVIEW

File: AzFilesHybrid/AzFilesHybrid.psm1 [hunk 1]
```diff
@@ -5468,6 +5470,313 @@ function New-AzDnsForwarder {
...
+{
+    <#
+    .SYNOPSIS
+    Maps local share level permissions to Azure RBAC's role assignments.
+    .DESCRIPTION
+    Local share level permissions applied on domain users will be mapped to Azure RBAC role assignments applied on cloud identity corresponding to domain users.
+    .OUTPUTS
+    Boolean, True, if ACL migration can happen without any errors if CommitChanges is set to False.
+    .EXAMPLE
+    PS C:\> ACLMigration -LocalSharename "<localsharename>" - Destinationshare "<destinationshharename>" -ResourceGroupName "<resourceGroupName>" -StorageAccountName "<storageAccountName>" -CommitChanges $False -StopOnAADUserLookupFailure $True -AutoFitSharePermissionsOnAAD $True
```
Comment (YunZhou-MS) [inline AzFilesHybrid/AzFilesHybrid.psm1:5553]: Remove the extra space

File: AzFilesHybrid/AzFilesHybrid.psm1 [hunk 2]
```diff
@@ -5468,6 +5470,313 @@ function New-AzDnsForwarder {
...
+         [Parameter(Mandatory=$true, Position=2, HelpMessage="Resource group name of storage account.")]
+         [string]$ResourceGroupName,
+
+         [Parameter(Mandatory=$true, Position=3, HelpMessage="Storage account name on Azure")]
+         [string]$StorageAccountName,
+
+         [Parameter(Mandatory=$true, Position=4, HelpMessage="If false, the tool just checks for possible errors and reports back without making any changes on the cloud.")]
+         [bool]$CommitChanges,
+
+         [Parameter(Mandatory=$false, Position=5, HelpMessage="If true, ACL migration will be stopped upon failure to identity local user on AAD")]
```
Comment (YunZhou-MS) [inline AzFilesHybrid/AzFilesHybrid.psm1:5572]: "to identity local user on AAD": identify?

File: AzFilesHybrid/AzFilesHybrid.psm1 [hunk 3]
```diff
@@ -5468,6 +5470,313 @@ function New-AzDnsForwarder {
...
+        Write-Verbose -Message "Mapping domain user/group - $account to its corresponding identity on Azure AAD"
+
+        #Geting the OID of domain user/group using its SID
+        try
+        {
+            $aadUser = Get-AzureADUser -Filter "OnPremisesSecurityIdentifier eq '$strSID'"
+        }
+        catch
+        {
+            throw
```
Comment (YunZhou-MS) [inline AzFilesHybrid/AzFilesHybrid.psm1:5629]: Is the try-catch needed if the error is always thrown out?

File: AzFilesHybrid/AzFilesHybrid.psm1 [hunk 4]
```diff
@@ -5468,6 +5470,313 @@ function New-AzDnsForwarder {
...
+            Write-Error -Message "The Storage Account doesn't exist. To create the Storage account and connect it to active directory, 
+                                    please follow the link https://docs.microsoft.com/en-us/azure/storage/files/storage-files-identity-auth-active-directory-enable" 
+                        -ErrorAction Stop
+        }
+
+        return $storageAccount
+    }
+}
+
+function ACLMigration
```
Comment (YunZhou-MS) [inline AzFilesHybrid/AzFilesHybrid.psm1:5543]: Need a better name, something like "Migrate-ShareAclToAzure"? Basically the name needs to conform to the powershell "Verb-Noun" convention and also be descriptive about what it is going to do.

File: AzFilesHybrid/AzFilesHybrid.psm1 [hunk 5]
```diff
@@ -5468,6 +5470,313 @@ function New-AzDnsForwarder {
 #region DFS-N cmdlets
 #endregion
 
+#region Share level permissions migration cmdlets
+function Get-StorageAccountDetails {
```
Comment (YunZhou-MS) [inline AzFilesHybrid/AzFilesHybrid.psm1:5474]: Is this supposed to be used by customers?
The name is a "Get-*" operation on an account but has a side effect of creating a share object, which is misleading. Need a better name. If this is just for testing purpose, I am not sure if we need this cmdlet.

Comment (chdevala) [reply AzFilesHybrid/AzFilesHybrid.psm1]: Removed this function. As you are saying it doing more than the name suggests

File: AzFilesHybrid/AzFilesHybrid.psm1 (hunk 4 above)
Comment (chdevala) [reply AzFilesHybrid/AzFilesHybrid.psm1]: Followed Verb-Noun convestion. Changed the name

File: AzFilesHybrid/AzFilesHybrid.psm1 (hunk 1 above)
Comment (chdevala) [reply AzFilesHybrid/AzFilesHybrid.psm1]: done

File: AzFilesHybrid/AzFilesHybrid.psm1 (hunk 2 above)
Comment (chdevala) [reply AzFilesHybrid/AzFilesHybrid.psm1]: done

File: AzFilesHybrid/AzFilesHybrid.psm1 (hunk 3 above)
Comment (chdevala) [reply AzFilesHybrid/AzFilesHybrid.psm1]: my bad. Missed this

**Pull request merged**